src/api/services/ListingItemActionService.ts

```diff
@@ -70,6 +72,67 @@ export class ListingItemActionService {
         const itemTemplateModel = await this.listingItemTemplateService.findOne(data.listingItemTemplateId, true);
         const itemTemplate = itemTemplateModel.toJSON();
 
+        const FRACTION_TO_COMPRESS_BY = 0.6;
+        const FRACTION_TO_RESIZE_IMAGE_BY = 0.6;
+        const MAX_SMSG_SIZE = 400000; // TODO: Give these proper values and move them out of here
+        const OVERHEAD_PER_SMSG = 0;
+        const OVERHEAD_PER_IMAGE = 0;
+        const MAX_RESIZES = 20;
+        const itemImages = itemTemplate.ItemInformation.ItemImages;
+        // ItemInformation has ItemImages, which is an array.
+        // Each element in ItemImages has an array ItemImageDatas.
+        const sizePerImage = (MAX_SMSG_SIZE - OVERHEAD_PER_SMSG) / itemImages.length - OVERHEAD_PER_IMAGE;
+        for (const i in itemImages) {
+            if (i) {
+                let resizedImage;
+                let indexOfData;
+                {
+                    let foundOriginal = false;
+                    const itemImage = itemImages[i];
+                    for (const k in itemImage.ItemImageDatas) {
+                        if (k) {
+                            if (itemImage.ItemImageDatas[k].imageVersion === 'ORIGINAL') {
+                                resizedImage = itemImage.ItemImageDatas[k].data;
+                                foundOriginal = true;
```

Comment:
the image data is moved to ItemImages.ItemImageDatas.ItemImageDataContent.data

Reply:
Thanks. Changed. Will commit soon